### DISCUSSION TITLE
A newbie's Feedback about the SDK

### DISCUSSION THREAD

**Discussion by MarioGK** [Azure/azure-cosmos-dotnet-v3#3844] [General]
Hello, i just started using CosmosDB and loved the concept of schemeless data, that supports SQL and the ease of scale down and up of CosmosDB, but the SDK seems to be old and missing some QoL features that i do not know why

Complaints:
- Dependency on Newtonsoft.Json while all other microsoft libraries and products are hard pushing System.Text.Json
- No support for IAsyncEnumerable (AFAIK), it would be an awesome feature that is easy to use instead of the Iterator
- You cannot simply get the raw json from the object, you need to serialize it to JObject or dynamic (This is my biggest complaint)
- I missed a feature to throttle requests, to give priority to certain requests than others
- Convert a IQueriable to SQL, i think CosmosDB has all the features to do this one, it just needs to be a public helper function or an extension

**Comment by gpuchtel** [comment]
The continued dependency on Newtonsoft if frustrating.
